src/main/scala/com/microsoft/hyperspace/logging/HyperspaceEvent.scala

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (2020) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.hyperspace.logging
+
+import org.apache.commons.lang.StringUtils
+import org.apache.spark.SparkContext
+
+import com.microsoft.hyperspace.index.IndexLogEntry
+
+/**
+ * Base class for all Hyperspace events.
+ */
+class HyperspaceEvent(
+    sparkUser: String = StringUtils.EMPTY,
+    appId: String = StringUtils.EMPTY,
+    appName: String = StringUtils.EMPTY) {
+  def dimensions: Seq[String] = Seq[String](sparkUser, appId, appName)
+}
+
+/**
+ * General index CRUD event.
+ *
+ * @param sparkUser user of spark application.
+ * @param appId spark application id.
+ * @param appName spark application name.
+ * @param indexes related indexes.
+ * @param message message about event.
+ */
+class HyperspaceIndexCRUDEvent(
+    sparkUser: String,
+    appId: String,
+    appName: String,
+    indexes: Seq[IndexLogEntry],
+    message: String = "")
+    extends HyperspaceEvent(sparkUser, appId, appName) {
+
+  override def dimensions: Seq[String] = {
+    commonDimensions ++ Seq("", "") ++ Seq(message)
+  }
```

Comment:
Confused about what's going on in this line.

Reply:
Can we just remove `dimensions` as suggested here: https://github.com/microsoft/hyperspace/pull/87#discussion_r453082594 ?

Reply:
done, thanks